airflow/dag_processing/manager.py

```diff
@@ -560,54 +505,92 @@ def _run_parsing_loop(self):
                     # only drives "max runs")
                     self.log.debug("BlockingIOError received trying to send DagParsingStat, ignoring")
 
-                if max_runs_reached:
-                    self.log.info(
-                        "Exiting dag parsing loop as all files have been processed %s times", self._max_runs
-                    )
-                    if span.is_recording():
-                        span.add_event(
-                            name="info",
-                            attributes={
-                                "message": "Exiting dag parsing loop as all files have been processed {self._max_runs} times"
-                            },
-                        )
-                    break
+            if self.max_runs_reached():
+                self.log.info(
+                    "Exiting dag parsing loop as all files have been processed %s times", self.max_runs
+                )
+                break
 
-                loop_duration = time.monotonic() - loop_start_time
-                if loop_duration < 1:
-                    poll_time = 1 - loop_duration
-                else:
-                    poll_time = 0.0
+            loop_duration = time.monotonic() - loop_start_time
+            if loop_duration < 1:
+                poll_time = 1 - loop_duration
+            else:
+                poll_time = 0.0
+
+    def _service_processor_sockets(self, timeout: float | None = 1.0):
+        """
+        Service subprocess events by polling sockets for activity.
+
+        This runs `select` (or a platform equivalent) to look for activity on the sockets connected to the
+        parsing subprocesses, and calls the registered handler function for each socket.
+
+        All the parsing processes socket handlers are registered into a single Selector
+        """
+        events = self.selector.select(timeout=timeout)
+        for key, _ in events:
+            socket_handler = key.data
+            need_more = socket_handler(key.fileobj)
+
+            if not need_more:
+                self.selector.unregister(key.fileobj)
+                key.fileobj.close()  # type: ignore[union-attr]
```

Comment:
This isn't a big func so not a big deal -- but thinking if we can re-use: `WatchedSubprocess._service_subprocess`